### PR TITLE
adding github pages docs publishing action

### PR DESCRIPTION
**Pull request by dylanbannon**
## what
* Adding a GitHub Action to publish terraform-aws-components docs to GitHub Pages.

## why
* This is a good way to have a dedicated docs site for terraform-aws-components.

## references
* Relates to https://github.com/cloudposse/docs/pull/524

### CODE REVIEW

File: .github/workflows/github-pages-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      GITHUB_PAGES_REPO: https://github.com/cloudposse/terraform-aws-components
+      GITHUB_PAGES_PUSH_BRANCH: docs
+      GITHUB_PAGES_DIRECTORY: github_pages
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Build Hugo Static Site"
+      uses: cloudposse/docs/actions/publish-github-pages@publish-gh-pages-python
+      with:
+        CSS: "body {\n>.__slackin {\n  z-index: 2;\n}\nfont-size: 1.1rem;\n}"
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:30]:
```suggestion
        CSS: |
          body {
            >.__slackin {
              z-index: 2;
            }
            font-size: 1.1rem;
          }
```

File: .github/workflows/github-pages-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        STAGING_DIR: /tmp/staging
+        CONTENT: docs,modules
+        DEBUG: True
+        HUGO_URL: https://cloudposse.github.io/terraform-aws-components
+        HUGO_PUBLISH_DIR: docs
+        HUGO_CONFIG: hugo.config.new.yml # this variable is the file output location, not the input template location
+        HTMLTEST_CONFIG: .htmltest.new.yml # this variable is the file output location, not the input template location
+        GITHUB_PAGES_PULL_BRANCH: github-pages-docs
+        GITHUB_PAGES_HUGO_PATH: /tmp/hugo
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:39]: Use lower-case with `with`

File: .github/workflows/github-pages-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - name: "Deploy Hugo Static Site to GitHub Pages"
+      uses: cloudposse/actions/github/git-push@0.30.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_ACTOR: ${{ secrets.GITHUB_ACTOR }}
+        GITHUB_REPOSITORY: env.GITHUB_PAGES_REPO
+        GIT_DIRECTORY: ./${{ env.GITHUB_PAGES_DIRECTORY }}/
+        GIT_BRANCH: ${{ env.GITHUB_PAGES_PUSH_BRANCH }}
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:47]: Lets move this to the composite action.

File: .github/workflows/github-pages-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - name: "Build Hugo Static Site"
+      uses: cloudposse/docs/actions/publish-github-pages@publish-gh-pages-python
+      with:
+        CSS: "body {\n>.__slackin {\n  z-index: 2;\n}\nfont-size: 1.1rem;\n}"
+        STAGING_DIR: /tmp/staging
+        CONTENT: docs,modules
+        DEBUG: True
+        HUGO_URL: https://cloudposse.github.io/terraform-aws-components
+        HUGO_PUBLISH_DIR: docs
+        HUGO_CONFIG: hugo.config.new.yml # this variable is the file output location, not the input template location
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:36]:
```suggestion
        STAGING_DIR: /tmp/staging
        IMPORT: "tutorials,howto,fundamentals"
        COLLATE: docs, modules
        DEBUG: True
        HUGO_URL: https://cloudposse.github.io/terraform-aws-components
        HUGO_PUBLISH_DIR: docs
        HUGO_CONFIG: hugo.config.new.yml # this variable is the file output location, not the input template location
```

File: .github/workflows/github-pages-deployment.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      GITHUB_PAGES_PUSH_BRANCH: docs
+      GITHUB_PAGES_DIRECTORY: github_pages
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Build Hugo Static Site"
+      uses: cloudposse/docs/actions/publish-github-pages@publish-gh-pages-python
+      with:
+        CSS: "body {\n>.__slackin {\n  z-index: 2;\n}\nfont-size: 1.1rem;\n}"
+        STAGING_DIR: /tmp/staging
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:31]:
```suggestion
```

File: .github/workflows/github-pages-deployment.yml (hunk 4 above)
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:36]:
```suggestion
```

File: .github/workflows/github-pages-deployment.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        CONTENT: docs,modules
+        DEBUG: True
+        HUGO_URL: https://cloudposse.github.io/terraform-aws-components
+        HUGO_PUBLISH_DIR: docs
+        HUGO_CONFIG: hugo.config.new.yml # this variable is the file output location, not the input template location
+        HTMLTEST_CONFIG: .htmltest.new.yml # this variable is the file output location, not the input template location
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:37]:
```suggestion
```

File: .github/workflows/github-pages-deployment.yml (hunk 3 above)
Comment (dylanbannon) [reply .github/workflows/github-pages-deployment.yml:53]: I don't think that's possible. My understanding of `composite` actions is that they allow you to run a series of shell scripts, but I don't think they currently allow you to call other github actions.
Here's an announcement of a third-party solution for this that seems to share my understanding: https://github.com/actions/runner/issues/646#issuecomment-790007107

File: .github/workflows/github-pages-deployment.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+defaults:
+  run:
+    shell: bash +e -x -o pipefail {0}
+
+jobs:
+  github-pages-deploy:
+    name: "Build Hugo Site and Deploy to GitHub Pages"
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAGES_REPO: https://github.com/cloudposse/terraform-aws-components
```
Comment (osterman) [inline .github/workflows/github-pages-deployment.yml:21]: Let's default this to the current repo

**Review by Benbentwo — APPROVED** [review]

**Comment by osterman** [comment]
superseded by https://github.com/cloudposse/docs/pull/550

**Pull request closed without merging**